hazelcast/src/main/java/com/hazelcast/query/impl/getters/JsonDataGetter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl.getters;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.hazelcast.nio.serialization.Data;
+
+import java.io.IOException;
+
+import static com.hazelcast.internal.serialization.impl.HeapData.HEAP_DATA_OVERHEAD;
+
+@SuppressWarnings("checkstyle:magicnumber")
+public final class JsonDataGetter extends AbstractJsonGetter {
+
+    public static final JsonDataGetter INSTANCE = new JsonDataGetter();
+
+    private JsonFactory factory = new JsonFactory();
```

Comment:
Can we have a static factory since the javadoc allows it? Maybe a single one in `AbstractJsonGetter `?

```
 * Factory instances are thread-safe and reusable after configuration
 * (if any). Typically applications and services use only a single
 * globally shared factory instance, unless they need differently
 * configured factories. Factory reuse is important if efficiency matters;
 * most recycling of expensive construct is done on per-factory basis.
```

Reply:
Ok I see there is a singleton for the `JsonGetter` and `JsonDataGetter` so this is a minor improvement.